libr/util/r_asn1.c

```diff
@@ -381,17 +380,16 @@ ut32 r_asn1_count_objects (const ut8 *buffer, ut32 length) {
 	object = NULL;
 	next = buffer;
 	end = buffer + length;
-	while (next >= buffer && next < end) {
+	while (next >= buffer && end > next ) {
 		object = asn1_parse_header (next, end - next);
 		if (!object || next == object->sector) {
-			R_FREE (object);
 			break;
 		}
 		next = object->sector + object->length;
 		counter++;
-		R_FREE (object);
+		RASN1_OBJ_FREE0 (object);
```

Comment:
only the first free requires to null the pointer to avoid the double free below, dont use macros for this